win32ss/gdi/ntgdi/region.c

```diff
@@ -3452,8 +3451,11 @@ REGION_SetPolyPolygonRgn(
         }
     }
 
-    REGION_FreeStorage(SLLBlock.next);
     REGION_PtsToRegion(numFullPtBlocks, iPts, &FirstPtBlock, prgn);
+    bResult = TRUE;
+
+Cleanup:
+    REGION_FreeStorage(SLLBlock.next);
 
     for (curPtBlock = FirstPtBlock.next; numFullPtBlocks-- > 0;)
```

Comment:
Side note: This for-loop is quite similar to the `REGION_FreeStorage` function -- excepting for the extra `numFullPtBlocks` decrement. I wonder whether that for-loop could be replaced with a FreeStorage call; also, I wonder what would be then the effect of not looking at the `numFullPtBlocks`.

Reply:
If we used C++, this would be a destructor.